### DISCUSSION TITLE
How to not touch Pipfile, but *only* Pipfile.lock?

### DISCUSSION THREAD

**Discussion by sbrandtb** [renovatebot/renovate#8775] [Request Help]
We are using pipenv and generally, all our direct dependencies are added as `*` version. Some cases have hard version constraints like `<2` because that would break our code.

Now, how to make renovate not touch dependencies in Pipfile at all, but only Pipfile.lock? This sounds like something super simple. I am probably just not searching for the right thing.

Bonus: It should still propose updates to `python_version`

**Comment by viceice** [comment]
https://docs.renovatebot.com/configuration-options/#lockfilemaintenance

**Reply by sbrandtb** [reply]
That will update the lockfile, but I explicitly want renovate bot to *not* touch the constraints in the Pipfile itself

**Reply by viceice** [reply]
Yes, the lockfile mainenance will not touch the `Pipfile` and will only update the `Pipfile.lock` . Isn't this what you want?

If not, please make your question more precise.

**Reply by sbrandtb** [reply]
That is what I want, but I seem to be missing something because it is still trying to change the `Pipfile`:

```toml
# Pipfile, simplified

[[source]]
name = "pypi"
url = "https://pypi.org/simple"
verify_ssl = true

[dev-packages]
pytest = "*"

[packages]
simple-rest-client = "<1"

[requires]
python_version = "3.8"
```

`renovate.json`:
```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "config:base",
    ":maintainLockFilesWeekly"
  ]
}
```

I merged the new `renovate.json`, closed the merge request that upgraded `simple-rest-client`, updated to latest (24.62.2) renovate bot, executed `renovate` - and it recreated the merge request that updates `simple-rest-client` to `<2`:

```diff
diff --git a/Pipfile b/Pipfile
index fc4bd71..c80f74e 100644
--- a/Pipfile
+++ b/Pipfile
@@ -33,7 +33,7 @@ aioboto3 = "*"
 tenacity = "*"
 dictlib = "*"
 # Version 1 switches from aiohttp to httpx. Breaks our tests with aioresponses.
-simple-rest-client = "<1"
+simple-rest-client = "<2"
 pyquery = "*"
 pendulum = "*"
```

**Reply by viceice** [reply]
use a package rule with `dependencyDashboardApproval` to so you can  manually allow pr creations.

https://docs.renovatebot.com/configuration-options/#dependencydashboardapproval

see sample:
https://github.com/renovatebot/.github/blob/13fe2cdee0f091b73fcf576c503b531dedc27246/default.json#L20-L25

```json
  {
      "description": "Require dashboard approval for major updates except Renovate",
      "excludePackagePatterns": ["^renovate(bot)?($|/)"],
      "matchUpdateTypes": ["major"],
      "dependencyDashboardApproval": true
    },
```

**Reply by rarkins** [reply]
Enabling lock file maintenance doesn't disable regular updates. They are not mutually exclusive

**Reply by sbrandtb** [reply]
@rarkins so it is not possible to make renovate bot not touch the requirements in `Pipfile` at all?

**Reply by viceice** [reply]
@sbrandtb I i told you, you need to add some package rules to disable the updates

```json
 {
      "description": "disable all pipenv updates",
      "matchManagers": ["pipenv"],
      "enabled": false
    },
```

**Reply by rarkins** [reply]
@viceice maybe needs matchUpdateTypes too?

**Reply by viceice** [reply]
not sure, maybe 😄

**Reply by rarkins** [reply]
```
{
      "description": "disable all pipenv updates",
      "matchManagers": ["pipenv"],
      "matchUpdateTypes": ["patch", "minor", "major"],
      "enabled": false
    },
```

We'd need a reproduction repo for anything more
